lib/protocol/SqlString.js

```diff
@@ -72,20 +73,41 @@ SqlString.arrayToList = function arrayToList(array, timeZone) {
 };
 
 SqlString.format = function format(sql, values, stringifyObjects, timeZone) {
-  values = values == null ? [] : [].concat(values);
+  if (values == null) {
+    return sql;
+  }
 
-  var index = 0;
-  return sql.replace(/\?\??/g, function(match) {
-    if (index === values.length) {
-      return match;
-    }
+  if (!(values instanceof Array)) {
```

Comment:
`instaceof` is twice as fast in V8 5.0 and historically has been at least 1.5x faster.
```
Array.isArray - true x 52,454,755 ops/sec ±0.90% (90 runs sampled)
instanceof Array - true x 103,139,337 ops/sec ±1.47% (93 runs sampled)
Array.isArray - false x 50,425,812 ops/sec ±0.80% (91 runs sampled)
instanceof Array - false x 97,070,525 ops/sec ±1.42% (86 runs sampled)
```

The only array `Array.isArray` will catch that `instanceof` won't is an array from a `vm`, but I'd assume most people pulling the Array constructor from a `vm` will be following [this snippet](https://gist.github.com/CrabDude/1100841) which lets `instanceof` work correctly.